codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -395,14 +386,11 @@ public void onRstStreamRead(ChannelHandlerContext ctx, int streamId, long errorC
             verifyPrefaceReceived();
 
             Http2Stream stream = connection.requireStream(streamId);
-            verifyRstStreamNotReceived(stream);
             if (stream.state() == CLOSED) {
                 // RstStream frames must be ignored for closed streams.
                 return;
             }
 
-            stream.resetReceived();
-
             listener.onRstStreamRead(ctx, streamId, errorCode);
```

Comment:
@Scottmitch it looks like the only place we're using this is [InboundHttp2ToHttpAdapter](https://github.com/netty/netty/blob/master/codec-http2/src/main/java/io/netty/handler/codec/http2/InboundHttp2ToHttpAdapter.java#L306).  Should this code be handling general stream closure rather than specifically listening for RST_STREAM?